### PR TITLE
Build & Test iOS integration test against repo

### PR DESCRIPTION
**Pull request by sunmou99**
Workflow to build & test iOS integration test
 - add build iOS framework against repo
 - change to build unsigned .ipa and delete signing code.
 - build vcpkg only for Desktop platform 
 - fix bug b/163338992
 - exclude [iOS,ubuntu-latest] [iOS,windows-latest] from matrix
 - install gcloud components "beta" 
 - add function to set bundle id.

### CODE REVIEW

**Comment by anonymous-akorn** [comment]
Could you add some context on this PR? It looks like you're merging one of your feature branches into another one of your feature branches, which doesn't require approval. Are you just looking for some early input on these particular changes?

**Comment by sunmou99** [comment]
Talked with akorn about the context on this PR.

File: scripts/gha/build_testapps.py
Comment (anonymous-akorn) [inline scripts/gha/build_testapps.py:190]: If any framework fails to build, the entire script will stop, which is not what we want: errors in iOS should not affect desktop or Android, and a failure for database should not affect auth. 

It would be better if the frameworks needed for a given API were built within the try block (line 268) for the iOS build of that platform.

**Review by anonymous-akorn — CHANGES_REQUESTED** [review]
Looks mostly good at a high level.

File: scripts/gha/integration_testing/xcodebuild.py
Comment (sunmou99) [reply scripts/gha/integration_testing/xcodebuild.py:31]: Will do

File: scripts/gha/integration_testing/xcodebuild.py
Comment (sunmou99) [reply scripts/gha/integration_testing/xcodebuild.py:58]: But it will looks inconsistent with other args.

File: scripts/gha/integration_testing/xcode_tool.rb
Comment (sunmou99) [reply scripts/gha/integration_testing/xcode_tool.rb:130]: Not all the xcode projects have this field set: 
https://github.com/firebase/firebase-cpp-sdk/blob/feature/gha-ios-integration-tests/messaging/integration_test/integration_test.xcodeproj/project.pbxproj#L206

I also remember FPL has a script to set PRODUCT_BUNDLE_IDENTIFIER in google3.

File: scripts/gha/build_testapps.py
Comment (sunmou99) [reply scripts/gha/build_testapps.py:423]: They are not the same path. I remember testapp_builder_dir is a subdirectory in project_dir.

File: scripts/gha/integration_testing/xcodebuild.py
Comment (anonymous-akorn) [reply scripts/gha/integration_testing/xcodebuild.py:58]: More readable, though. PEP8 explicitly prescribes doing this to avoid escape characters in the string:
https://www.python.org/dev/peps/pep-0008/#string-quotes

File: scripts/gha/test_lab.py
Comment (anonymous-akorn) [inline scripts/gha/test_lab.py:298]: Should the quiet flag be added to the Android command as well?

File: scripts/gha/integration_testing/xcodebuild.py
Comment (anonymous-akorn) [inline scripts/gha/integration_testing/xcodebuild.py:88]: There are a few undocumented assumptions/behaviours here, which should either be explained in the docstring or exposed as arguments.

- There's a subdirectory called "Debug-iphoneos" in the build dir, which implicitly assumes we ran xcodebuild with "Debug" configuration.
- "build_dir" is the directory containing Debug-iphoneos, and not Debug-iphoneos itself.
- The resulting ipa will be called "integration_test.ipa", and will be placed within Debug-iphoneos, not build_dir.

File: scripts/gha/build_testapps.py
Comment (anonymous-akorn) [inline scripts/gha/build_testapps.py:189]: This variable is misleading - it will only be set to True if we're building from repo **and** building for iOS. If someone decides to add any logic that depends on whether we're building from repo for Desktop or Android, they'll run into problems.

It would make more sense to move the check out of _generate_makefiles_from_repo, and only run that method if github_repo is true and we're building for iOS. We should also log it (whether we've detected that we're building from repo) for debugging purposes.

File: scripts/gha/build_testapps.py
Comment (anonymous-akorn) [inline scripts/gha/build_testapps.py:399]: Is this identical comment supposed to be before both _build_ios_framework_from_repo and _generate_makefiles_from_repo?

**Review by anonymous-akorn — CHANGES_REQUESTED** [review]

File: scripts/gha/test_lab.py
Comment (sunmou99) [reply scripts/gha/test_lab.py:298]: There is no interactive prompts for android run. But probably yes.

File: scripts/gha/integration_testing/xcodebuild.py
Comment (sunmou99) [reply scripts/gha/integration_testing/xcodebuild.py:88]: Will add document, and change the variables' name.

File: scripts/gha/test_lab.py
Comment (sunmou99) [reply scripts/gha/test_lab.py]: Change the components installation into yml.

File: .github/workflows/integration_tests.yml
Comment (anonymous-akorn) [inline .github/workflows/integration_tests.yml:124]: Is there any harm in putting this command in test_lab.py? This will be needed both locally and on GHA, so it seems sensible to put it in the script instead.

**Review by anonymous-akorn — CHANGES_REQUESTED** [review]
Looks like there's a merge conflict? Also some of these files look fairly outdated, like they're missing changes from dev. Might need to merge from dev.

File: .github/workflows/integration_tests.yml
Comment (sunmou99) [reply .github/workflows/integration_tests.yml:124]: There is no harm putting it in test_lab.py, except it require more code and code looks not good(an extra variable to make it only for iOS) in test_lab.py.
To run it locally, there is interactive prompts. Just input "Y" to install the components.

**Comment by sunmou99** [comment]
> Looks like there's a merge conflict? Also some of these files look fairly outdated, like they're missing changes from dev. Might need to merge from dev.

Yes. I noticed it too. But I have resolved the merge conflict more than 3 times. So, I decide not to resolve it until feature/package-ios-frameworks get merged. 
The files are rebased/merge from dev this Monday, not quite outdated. Again, I will rebased/merge from dev after feature/package-ios-frameworks get merged.

File: .github/workflows/integration_tests.yml
Comment (anonymous-akorn) [reply .github/workflows/integration_tests.yml:124]: Having to include this step in every workflow that uses the script is a significant downside. 

To avoid having to add another flag, the logic can choose to run this command if we find any iOS testapps, i.e. could set a boolean here:
https://github.com/firebase/firebase-cpp-sdk/blob/dev/scripts/gha/test_lab.py#L119

Comment (sunmou99) [reply .github/workflows/integration_tests.yml:124]: This is what I mean the code looks not good. It is because we build iOS and then we got IPAs. It's strange that we say we have IPAs, thus we are building for iOS (especially in a loop).
1. To write it into the script, I prefer adding a flag to test_lab.py itself. Like: --p ${{ matrix.target_platform }}
2. Or we could put it into step "Install Cloud SDK for mobile integration tests":
`      - name: Install Cloud SDK for mobile integration tests
        if: matrix.target_platform != 'Desktop' && !cancelled()
        uses: GoogleCloudPlatform/github-actions/setup-gcloud@master 
        run: |
          gcloud --quiet components install beta
`
   Thus, this step installs both gcloud and gcloud compoent beta. Though it installs gcloud compoent beta for Android, it only takes about 30 seconds.

Comment (anonymous-akorn) [reply .github/workflows/integration_tests.yml:124]: The logic is "we passed a directory with testapps to this tool; one or more of these testapps is for iOS; thus we need to install tools for testing iOS". That's intuitive, no? 

The flag would add complexity to its usage, and is redundant since we need to install these tools if and only if the directory of testapps include iOS testapps, information we already have available to us in the context of the tool.

**Review by anonymous-akorn — APPROVED** [review]

**Pull request merged**
